contracts/interchain-token-service/tests/flow_limit.rs

```diff
@@ -0,0 +1,290 @@
+mod utils;
+
+use axelar_gateway::types::Message as GatewayMessage;
+use axelar_soroban_std::{assert_contract_err, assert_invoke_auth_ok, events, traits::BytesExt};
+use interchain_token_service::{
+    error::ContractError,
+    event::FlowLimitSetEvent,
+    types::{HubMessage, InterchainTransfer, Message},
+    InterchainTokenServiceClient,
+};
+use soroban_sdk::testutils::{Address as _, Ledger as _};
+use soroban_sdk::{vec, xdr::ToXdr, Address, Bytes, BytesN, Env, String, Vec};
+use utils::{
+    approve_gateway_messages, register_chains, setup_env, setup_gas_token, setup_its_token,
+    HUB_CHAIN,
+};
+
+const TEST_FLOW_LIMIT: Option<i128> = Some(1000);
+const EPOCH_TIME: u64 = 6 * 60 * 60;
+
+fn setup_flow_limit(env: &Env, client: &InterchainTokenServiceClient) -> (BytesN<32>, Address) {
+    let supply = i128::MAX;
+    let deployer = Address::generate(env);
+    let token_id = setup_its_token(env, client, &deployer, supply);
+
+    client
+        .mock_all_auths()
+        .set_flow_limit(&token_id, &TEST_FLOW_LIMIT);
```

Comment:
```suggestion
        .set_flow_limit(&token_id, &Some(TEST_FLOW_LIMIT));
```
remove the `Some` from the constant to avoid calling `unwrap`s later